File: Back to Basics Approach 3rd Edition/cse_143/cs2_sections/section01_arrayList/indexOf.java
package cse_143.cs2_sections.section01_arrayList;

/**
 * https://practiceit.cs.washington.edu/problem/view/cs2/sections/arrays/indexOf
 *
 * @author Alex Golub
 * @since 19-Apr-21 7:58 PM
 */
class indexOf {
  public static int indexOf(int[] arr, int value) {
    for (int i = 0; i < arr.length; i++) {
      if (arr[i] == value) {
        return i;
      }
    }
    
    return -1;
  }
}

File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/javascript/recursion/indexOf.js
/**
 * https://www.codestepbystep.com/problem/view/javascript/recursion/indexOf
 *
 * @author Alex Golub
 * @since 08-May-21, 1:30 PM
 */
function indexOf(src, tgt) {
  if (tgt.length > src.length) return -1;
  if (src.startsWith(tgt)) return 0;
  let nextIdx = indexOf(src.substring(1), tgt);
  return nextIdx === -1 ? -1 : 1 + nextIdx;
}
